### PR TITLE
Feature/#12896 Capture & Save Card Data

### PR DESCRIPTION
**Pull request by ovidiuurotaru**

### CODE REVIEW

File: ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/OfflineModeKeychainStorage.swift [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+            currentSavedItems?.forEach({ item in
+                let data = item as? Data
+                if let newData = data {
+                    if let decryptedData = try? ClearentCryptor.decrypt(encryptionKey: encryptionKey, encryptedContent: newData) {
+                        let decodedResponse = try? JSONDecoder().decode(OfflineTransaction.self, from: decryptedData)
+                        if let decodedResponse = decodedResponse {
+                            result.append(decodedResponse)
+                        }
+                    }
+                }
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/OfflineModeKeychainStorage.swift:71]:
```suggestion
                if let newData = data,
                   let decryptedData = try? ClearentCryptor.decrypt(encryptionKey: encryptionKey, encryptedContent: newData,
                   let decodedResponse = try? JSONDecoder().decode(OfflineTransaction.self, from: decryptedData) {
                        result.append(decodedResponse)
                   }
```

File: ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/OfflineModeKeychainStorage.swift [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        currentSavedItems?.forEach({ item in
+            let data = item as? Data
+            if let newData = data {
+                if let decryptedData = try? ClearentCryptor.decrypt(encryptionKey: encryptionKey, encryptedContent: newData) {
+                    let decodedResponse = try? JSONDecoder().decode(OfflineTransaction.self, from: decryptedData)
+                    if let decodedResponse = decodedResponse {
+                        result.append(decodedResponse)
+                    }
+                }
+            }
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/OfflineModeKeychainStorage.swift:102]:
```suggestion
            if let newData = data,
               let decryptedData = try? ClearentCryptor.decrypt(encryptionKey: encryptionKey, encryptedContent: newData),
               let decodedResponse = try? JSONDecoder().decode(OfflineTransaction.self, from: decryptedData) {
                  result.append(decodedResponse)
            }
```

File: ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift [hunk 3]
```diff
@@ -190,20 +215,24 @@ public final class ClearentWrapper : NSObject {
...
         
         self.saleEntity = saleEntity
-
         if shouldDisplayConnectivityWarning(for: .payment) { return }
-
-        if let manualEntryCardInfo = manualEntryCardInfo {
-            manualEntryTransaction(cardNo: manualEntryCardInfo.card, expirationDate: manualEntryCardInfo.expirationDateMMYY, csc: manualEntryCardInfo.csc)
+        
+        if isManualTransaction {
+            // If offline mode is on
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift:230]: will this be replaced with the actual flag as part of a different story?

File: ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/OfflineModeKeychainStorage.swift [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+                }
+            })
+        }
+
+        return result
+    }
+
+    /// Fetches the saved data, serches for matching item , replaces the item  and saves the data, returns an error if the item is not found
+    /// If succesfull it will overwrite the data
+    func updateTransaction(transaction: OfflineTransaction) -> TransactionStoreStatus {
```
[the remark this answers is not in the document]
Comment (ovidiuurotaru) [reply ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/OfflineModeKeychainStorage.swift:76]: Should be fixed with the latest commit.

File: ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/OfflineModeKeychainStorage.swift [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+                }
+            }
+
+            return saveOfflineTransactionArray(offlineTransactions: currentSavedItems)
+        }
+
+        return response
+    }
+
+    func deleteTransactionWith(id: String) -> TransactionStoreStatus {
```
[the remark this answers is not in the document]
Comment (ovidiuurotaru) [reply ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/OfflineModeKeychainStorage.swift:119]: Should be fixed with the latest commit.

File: ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift (hunk 3 above)
Comment (ovidiuurotaru) [reply ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift:230]: Yes we should have a story to setup everything correctly, for now we do not have a UI for enabling disabling the offline mode.

File: ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/OfflineModeKeychainStorage.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (ovidiuurotaru) [reply ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/OfflineModeKeychainStorage.swift:76]: I simplified the update and delete operations.

**Pull request merged**